### PR TITLE
Replace mpqapi.h with the MpqWriter class

### PR DESCRIPTION
**Pull request by glebm**
This is a cleanup, heading in the direction of allowing us to reuse some of the code between reading and writing MPQs.

### CODE REVIEW

File: Source/utils/mpq_writer.cpp [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+		pbData += len;
+		len = PkwareCompress(mpqBuf, len);
+		if (!stream_.Write(reinterpret_cast<const char *>(&mpqBuf[0]), len))
+			return false;
+		sectoroffsettable[curSector++] = SDL_SwapLE32(destsize);
+		destsize += len; // compressed length
+		if (dwLen > SectorSize)
+			dwLen -= SectorSize;
+		else
+			break;
```
Comment (AJenbo) [inline Source/utils/mpq_writer.cpp:409]:
```suggestion
		if (dwLen <= SectorSize)
			break;

		dwLen -= SectorSize;
```

**Review by AJenbo — COMMENTED** [review]
Sorry I never got around to cleaning much of this :)

**Review by AJenbo — APPROVED** [review]

**Comment by glebm** [comment]
Squashed the last commit, needs a re-approval

**Review by AJenbo — APPROVED** [review]
Thanks that saves me the trouble :)

**Pull request merged**